pkg/crc/constants/constants_windows.go

```diff
@@ -12,3 +17,28 @@ const (
 )
 
 var TrayBinaryPath = filepath.Join(CrcBinDir, "tray-windows", TrayBinaryName)
+
+func GetHomeDir() string {
+	// if not running in interactive mode, it is the daemon service
+	// in that case we have the user's home directory as the third
+	// argument in the service executable invocation
+	isIntSess, err := svc.IsAnInteractiveSession()
+	if err == nil {
+		if !isIntSess && len(os.Args) > 2 {
+			return os.Args[2]
+		}
+	}
```

Comment:
I don't think this should go in such a generic function. If someone reuse `GetHomeDir` for non-tray code in other parts of CRC, they probably won't be expecting to sometimes get the 2nd cmd line arg as a home dir value. Maybe set an env var with argv[2] early in `crc daemon` run, and use that here?

Reply:
So, argv[2] will only be returned if the service is not running as an interactive process, and provided `crc` is a cli tool, this is not likely to occur outside of the tray scenario.

I actually tried using a variable to store `argv[2]` then using it in the constants pkg, but that creates import cycle..

Reply:
I would suggest to write logs when this happens... this can be, as said, very confusing... especially when debugging.

also, would it be better to use the username and reconstruct the path when actually needed? the user might have moved his path... or does this get properly cleaned with `crc cleanup` ?

Reply:
Yes, `crc cleanup` will remove the service..

Reply:
> I would suggest to write logs when this happens... this can be, as said, very confusing... especially when debugging.

we are depending on `svc.IsAnInteractiveSession` to behave properly, in normal situations the `os.Args[2]` won't be considered, but only when running as a service in windows. even if we put logs here it doesn't help us to debug the normal working of crc.

Reply:
> we are depending on svc.IsAnInteractiveSession to behave proper

Right, but we have no way to understand what happened. So I suggest that logging can help to understand if an interactive session and therefore the parameter was used. The parameter would have to be properly described, including possible localized characters to behave as expected.

if an error occurs, I would like to understand if this was due to the early return as an interactivesession or arg value was used or the regular flow.

Reply:
> I actually tried using a variable to store argv[2] then using it in the constants pkg, but that creates import cycle..
In `cmd/daemon.go`, call something like `constants.SetHomeDir(argv[2])` ? This should avoid the import cycle? (even if it's ugly to have a setter for constants :( )

Reply:
he wanted to avoid storing the value...

at the moment the value `argv[2]` does not convey any meaning. the problem is where to place such logic.